drivers/modem/modem_iface_uart.c

```diff
@@ -0,0 +1,150 @@
+/** @file
+ * @brief interface for modem context
+ *
+ * UART-based modem interface implementation for modem context driver.
+ */
+
+/*
+ * Copyright (c) 2019 Foundries.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(modem_iface_uart, CONFIG_MODEM_LOG_LEVEL);
+
+#include <kernel.h>
+#include <drivers/uart.h>
+
+#include "modem_context.h"
+#include "modem_iface_uart.h"
+
+/**
+ * @brief  Drains UART.
+ *
+ * @note   Discards remaining data.
+ *
+ * @param  *iface: modem interface.
+ *
+ * @retval None.
+ */
+static void modem_iface_uart_flush(struct modem_iface *iface)
+{
+	u8_t c;
+
+	while (uart_fifo_read(iface->dev, &c, 1) > 0) {
+		continue;
+	}
+}
+
+/**
+ * @brief  Modem interface interrupt handler.
+ *
+ * @note   Fills interfaces ring buffer with received data.
+ *         When ring buffer is full the data is discarded.
+ *
+ * @param  *uart_dev: uart device.
+ *
+ * @retval None.
+ */
+static void modem_iface_uart_isr(struct device *uart_dev)
+{
+	struct modem_context *ctx;
+	struct modem_iface_uart_data *data;
+	int rx = 0, ret;
+
+	/* lookup the modem context */
+	ctx = modem_context_from_iface_dev(uart_dev);
+	if (!ctx || !ctx->iface.iface_data) {
+		return;
+	}
+
+	data = (struct modem_iface_uart_data *)(ctx->iface.iface_data);
+	/* get all of the data off UART as fast as we can */
+	while (uart_irq_update(ctx->iface.dev) &&
+	       uart_irq_rx_ready(ctx->iface.dev)) {
+		rx = uart_fifo_read(ctx->iface.dev,
+				    data->isr_buf, data->isr_buf_len);
+		if (rx > 0) {
```

Comment:
I'd rather break if (rx == 0), less imbrication of if/if/if

Reply:
will fix.

Reply:
@tbursztyka actually, a break on (rx == 0) would change the current logic in 2 ways:
- `uart_fifo_read` can return an error (<0) so that needs to be handled as well
- current logic doesn't break the loop if rx is <=0, it re-checks `uart_irq_update` and `uart_irq_rx_ready`.

Instead I'll change to the following which preserves the logic:
```
               if (rx <= 0) {
                       continue;
               }
```